static/js/data_parser.js

```diff
@@ -50,34 +33,39 @@ function updateGPUData(data, currentUser) {
             if (!gpuRow.hasClass("alert-danger")) {
                 gpuRow.addClass("alert-danger");
             }
+            any_gpu_failed = true;
         } else {
             if (gpuRow.hasClass("alert-danger")) {
                 gpuRow.removeClass("alert-danger");
             }
         }
-        showCorrectButton(gpuRow, gpu, currentUser);
     }
-}
 
-function setupActionButtons() {
-    $(".action-button").on('click', function (){
-        const $this = $(this);
-        const url = $this.data('action');
-        const tagData = $this.data();
-        const postData = Object.keys(tagData)
-            .filter(key => key !== 'action')
-            .reduce((obj, key) => {
-                obj[key] = tagData[key];
-                return obj;
-            }, {});
+    const deviceHeading = $('#' + data.name + '_heading');
+    const deviceHeadingButton = deviceHeading.find('device-heading-btn')
+    if (any_gpu_in_use) {
```

Comment:
you could handle those two cases in a loop if you want, would help to remove some code duplication.

Reply:
You mean, handling this in *the* loop above, right?

Reply:
No :sweat_smile: , I meant you could handle the `any_gpu_in_use` and `any_gpu_failed` cases in a loop. The code is quite similar for both things only the variable and the `alert-x` is different.